proxy/pkg/zdmproxy/clientconn.go

```diff
@@ -238,12 +246,12 @@ func checkProtocolError(f *frame.RawFrame, connErr error, protocolErrorOccurred
 		}
 		rawProtocolErrResponse, err := generateProtocolErrorResponseFrame(streamId, protocolErrMsg)
 		if err != nil {
-			return nil, fmt.Errorf("could not generate protocol error response raw frame (%v): %v", protocolErrMsg, err)
+			return nil, fmt.Errorf("could not generate protocol error response raw frame (%v): %v", protocolErrMsg, err), errorCode
 		} else {
-			return rawProtocolErrResponse, nil
+			return rawProtocolErrResponse, nil, errorCode
 		}
 	} else {
-		return nil, connErr
+		return nil, connErr, errorCode
```

Comment:
`return nil, connErr, 0` here because `errorCode` is being set in the first if even if `protocolErrMsg` is `nil`